### ISSUE TITLE
[BUG] Opening a unix fifo in "write" mode throws an exception

### ISSUE THREAD

**Issue by diocletiann** [modularml/mojo#3208]
### Bug description

I'm not sure why this is failing, mentioned it on Discord and was asked to open an issue:
```
$ mojo run src/main.mojo
Unhandled exception caught during execution: unable to remove existing file /tmp/mojo.fifo
mojo: error: execution exited with a non-zero result: 1
```
This is the pipe:
```
$ ll /tmp/mojo.fifo
prwxrwxrwx  1 adi  wheel     0B Jul  9 14:12 /tmp/mojo.fifo|
```

### Steps to reproduce

```
$ mkfifo -m 777 /tmp/mojo.fifo
```

```mojo
fn main() raises:
    var handle = open("/tmp/mojo.fifo", "w")
```

### System information

```shell
MacOS 14.3.1
mojo 2024.7.905 (e860c551)
modular 0.8.0 (39a426b5)
```

**Comment by ematejska** [comment]
Looks like we're attempting to clear the the file before writing.  This reproduces if the file is in a location that's owned by root but doesn't reproduce if the file is in a location owned by the user.

**Comment by diocletiann** [comment]
> Looks like we're attempting to clear the the file before writing. This reproduces if the file is in a location that's owned by root but doesn't reproduce if the file is in a location owned by the user.

Yeah, I tried `open()` with a text file and it deleted the file before I could even be read.

**Comment by ematejska** [comment]
Does that happen when you open it for reading with "r".  I would expect that to happen when it's open for writing because the file is cleared out before writing happens….I believe that's how it works i Python as well.

**Comment by diocletiann** [comment]
"r" opens the file without data loss.
